src/clients/SpokePoolClient.ts

```diff
@@ -745,19 +721,12 @@ export class SpokePoolClient extends BaseAbstractClient {
         let deposit: DepositWithBlock;
 
         // Derive and append the common properties that are not part of the onchain event.
-        const { quoteBlock: quoteBlockNumber, realizedLpFeePct } = dataForQuoteTime[index];
-        if (this.isV3DepositEvent(event)) {
-          deposit = { ...(rawDeposit as V3DepositWithBlock), originChainId: this.chainId };
-          deposit.realizedLpFeePct = undefined;
-          deposit.quoteBlockNumber = quoteBlockNumber;
-          if (deposit.outputToken === ZERO_ADDRESS) {
-            deposit.outputToken = this.getDestinationTokenForDeposit(deposit);
-          }
-        } else {
-          deposit = { ...(rawDeposit as V2DepositWithBlock) };
-          deposit.realizedLpFeePct = realizedLpFeePct;
-          deposit.quoteBlockNumber = quoteBlockNumber;
-          deposit.destinationToken = this.getDestinationTokenForDeposit(deposit);
+        const { quoteBlock: quoteBlockNumber } = dataForQuoteTime[index];
+        deposit = { ...(rawDeposit as V3DepositWithBlock), originChainId: this.chainId };
+        deposit.realizedLpFeePct = undefined;
```

Comment:
This should probably also be removed as part of this work, assuming it doesn't have too much contagion.

Reply:
I'll follow up on this one once I flush the existing queue of PRs.